src/app/news.service.ts

```diff
@@ -10,16 +10,16 @@ export class NewsService {
   private _news$ = new Subject<string>();
   private _breakingNews$ = new Subject<string>();
 
-  news$ = this._news$.pipe(storeNews());
-  breakingNews$ = this._breakingNews$.pipe(storeNews());
+  news$ = this._news$.pipe(storeNews(), shareReplay());
+  breakingNews$ = this._breakingNews$.pipe(storeNews(), shareReplay());
 
-  allNews$ = combineLatest(
+  allNews$ = merge(
     // combineLatest will start emitting when all of it's streams will emit at least one time
     // that is why startWith() operator is required here - we want to emit normal news
     // even when there were no breaking news yet and vice versa
-    this.breakingNews$.pipe(startWith([])),
-    this.news$.pipe(startWith([]))
-  ).pipe(map(([breaking, news]) => [...breaking, ...news]));
+    this.breakingNews$,
+    this.news$
+  ).pipe(scan((breaking, news) => [...breaking, news[news.length - 1]], []));
```

Comment:
I believe this is working, however you already have the operator `storeNews()` that is aggregating all the news from single into an array and then here you again splitting it into single items and re-aggregating.
It seems like there is a more optimal way if you slightly restructure the streams.